src/coreclr/vm/debugdebugger.cpp

```diff
@@ -901,6 +852,31 @@ extern "C" void QCALLTYPE DebugDebugger_CustomNotification(QCall::ObjectHandleOn
     END_QCALL;
 #endif // DEBUGGING_SUPPORTED
 }
+
+extern "C" BOOL QCALLTYPE DebugDebugger_IsLoggingHelper()
+{
+    QCALL_CONTRACT_NO_GC_TRANSITION;
+
+#ifdef DEBUGGING_SUPPORTED
+    if (CORDebuggerAttached())
```

Comment:
So `CORDebuggerAttached()` makes a call to `RtlDllShutdownInProgress()` on Windows, but on non-Windows just does a `BOOL` check. The call into `RtlDllShutdownInProgress()` does give me pause. At the moment the implementation is trivial and I can "sign off" saying it is okay for `SuppressGCTransition`, but I'm not sure if this is something we want to codify for perhaps little gain.

@jkotas What do you think about relying on `RtlDllShutdownInProgress()`'s current implementation on Windows? If there is any concern, perhaps we shouldn't mark these functions as `SuppressGCTransition`.

Reply:
Historically, shutdown has been a complicated unreliable mess. We have been gradually fixing shutdown bugs by deleting a lot of code and simplifying the shutdown sequence.

Most places that call `IsAtProcessExit` should not be calling it. `CORDebuggerAttached` is probably one of those places.

Reply:
Good enough for me. Let's keep it as-is.